chore: add rust-toolchain.toml

Vid201 opened the pull request:
This PR adds rust toolchain file - it would be good to have fixed Rust version because there can be some problem with deps and Rust versions in the future when working with zkVMs (SP1 ...)

jihoonsong commented:
We can bump rust-version in cargo.toml (MSRV) to 1.85.0 if it doesn't break zkVM prototype cc @unnawut @syjn99

Vid201 commented on README.md:
```diff
@@ -2,7 +2,6 @@
 
 [![Telegram](https://img.shields.io/badge/Telegram-2CA5E0?style=for-the-badge&logo=telegram&logoColor=white)][tg-url]
 
-
```
Sorry about that, this was auto formatter for markdown. Some empty lines were removed, I can remove the changes if necessary

jihoonsong replied:
Yes, please remove all changes in readme.

hopinheimer commented:
`1.85.0` does bring some interesting features on the table, given I hope we don't run into issues especially because it was stabilised 2 weeks back. do you think we should wait for others dependencies to come to pace?

ref : https://bertptrs.nl/2025/02/23/rust-edition-2024-annotated.html

again I'm no expert cc @jun @unnawut

unnawut commented:
I'm using 1.85.0 locally actually, so I have no issue with this. Thanks for flagging!

unnawut commented:
Also cc: @KolbyML and @Kayden-ML who're working a lot on this repo

unnawut commented:
I just approved the github actions, the build is failing for Risc0 and SP1 (USA targets not found)

jihoonsong commented:
rust-toolchain.toml overrides succinct and risc0 toolchain and causes the CI issue. This uses command that specifies correct toolchain. Could you please change .github/workflows/riscv.yml with this? @Vid201 

```
name: RISC-V

on:
  push:
    branches: [ "master" ]
  pull_request:
    branches: [ "master" ]

env:
  CARGO_TERM_COLOR: always

jobs:
  build-sp1:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout Repository
      uses: actions/checkout@v4

    - name: Install toolchain
      run: |
        curl -L https://sp1up.succinct.xyz | bash
        /home/runner/.sp1/bin/sp1up
        rustup default succinct

    - name: Build SP1
      run: cargo +succinct build --target=riscv32im-succinct-zkvm-elf -p ream-consensus -p ream-bls

  build-risc0:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v4

    - name: Install toolchain
      run: |
        curl -L https://risczero.com/install | bash
        /home/runner/.risc0/bin/rzup install
        cargo risczero install
        rustup default risc0
        
    - name: Build Risc0
      run: cargo +risc0 build --target=riscv32im-risc0-zkvm-elf -p ream-consensus -p ream-bls

```

Vid201 commented:
@jihoonsong @unnawut updated

jihoonsong commented:
It turned out that we can't bump our rust-rversion to 1.85.0. RISC0 uses 1.81.1 explicitly.

`Update guest code to use Rust toolchain version 1.81.0.` [Source](https://github.com/risc0/risc0/blob/main/CHANGELOG.md)

jihoonsong commented:
@Vid201  Sorry that you've already made change but could you please do this following:
- Revert bumping rust-version in Cargo.toml
- Update ricsv.yml (I've edited it right after you applied it, sorry. I removed `rustup default TOOLCHAIN` lines as we don't need it anymore)

Vid201 commented:
> @Vid201 Sorry that you've already made change but could you please do this following:
> 
> * Revert bumping rust-version in Cargo.toml
> * Update ricsv.yml (I've edited it right after you applied it, sorry. I removed `rustup default TOOLCHAIN` lines as we don't need it anymore)

updated again :)

jihoonsong reviewed (approved):
All CIs work like a charm. Thanks a lot!

KolbyML commented:
I don't think there is a reason to specify a toolchain file https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file Here is some additional context from Reth/Lighthouse folks https://github.com/paradigmxyz/reth/issues/2821
